kimp/src/kimp/__main__.py

```diff
@@ -79,20 +80,35 @@ def main() -> None:
 def exec_run(
     input_file: str,
     definition_dir: str,
+    input_term: str | None = None,
     output: str = 'none',
     ignore_return_code: bool = False,
     **kwargs: Any,
 ) -> None:
     krun_output = KRunOutput[output.upper()]
 
-    definition_dir = str(find_definiton_dir('llvm'))
-
-    kimp = KIMP(definition_dir, definition_dir)
+    imp_parser = None
+    if definition_dir is None:
+        definition_dir_path = find_definiton_dir('llvm')
+        imp_parser = definition_dir_path / 'parser_Stmt_STATEMENTS-SYNTAX'
+        if not imp_parser.is_file():
+            imp_parser = gen_glr_parser(
+                imp_parser, definition_dir=definition_dir_path, module='STATEMENTS-SYNTAX', sort='Stmt'
+            )
+    else:
+        definition_dir_path = Path(definition_dir)
+    kimp = KIMP(definition_dir_path, definition_dir_path, imp_parser)
 
     try:
-        proc_res = kimp.run_program(input_file, output=krun_output)
-        if output != KAstOutput.NONE:
-            print(proc_res.stdout)
+        with NamedTemporaryFile(mode='w') as f:
+            temp_file = Path(f.name)
+            if input_term is not None:
+                temp_file.write_text(input_term)
+            else:
+                temp_file.write_text(Path(input_file).read_text())
+            proc_res = kimp.run_program(temp_file, output=krun_output)
```

Comment:
Of course the `run_program` code is still going to make a second copy of the file... but we don't have to fix that in this PR.